src/NHibernate.Test/NHSpecificTest/NH3918/FixtureByCode.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Expressions;
+using NHibernate.Cfg.MappingSchema;
+using NHibernate.Linq;
+using NHibernate.Mapping.ByCode;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.NH3918
+{
+	public class ByCodeFixture : TestCaseMappingByCode
+	{
+		protected override HbmMapping GetMappings()
+		{
+			var mapper = new ModelMapper();
+			mapper.Class<Owner>(rc =>
+			{
+				rc.Id(x => x.Id, m => m.Generator(Generators.GuidComb));
+				rc.Property(x => x.Name);
+			});
+			mapper.Class<Entity>(rc =>
+			{
+				rc.Id(x => x.Id, m => m.Generator(Generators.GuidComb));
+				rc.Property(x => x.Name);
+				rc.ManyToOne(m => m.Owner, m =>
+				{
+					m.Column("OwnerId");
+				});
+			});
+
+			return mapper.CompileMappingForAllExplicitlyAddedEntities();
+		}
+
+		protected override void OnSetUp()
+		{
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				var bob =  CreateOwner(session, "Bob");
+				var carl = CreateOwner(session, "Carl");
+				var doug = CreateOwner(session, "Doug");
+
+				CreateEntity(session, "Test 1", bob);
+				CreateEntity(session, "Test 2", carl);
+				CreateEntity(session, "Test 3", doug);
+				CreateEntity(session, "Test 4", bob);
+				CreateEntity(session, "Test 5", carl);
+				CreateEntity(session, "Test 6", doug);
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+		protected Owner CreateOwner(ISession session, string name)
+		{
+			var t = new Owner { Name = name };
+			session.Save(t);
+			return t;
+		}
+
+		protected void CreateEntity(ISession session, string name, Owner owner)
+		{
+			var t = new Entity
+			{
+				Name = name,
+				Owner = owner,
+			};
+			session.Save(t);
+		}
+
+		protected override void OnTearDown()
+		{
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				session.Delete("from Entity");
+				session.Delete("from Owner");
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+		[Test]
+		public void EntityComparisonTest()
+		{
+			using (ISession session = OpenSession())
+			using (session.BeginTransaction())
+			{
+				var bob = session.Query<Owner>().Single(o => o.Name == "Bob");
+
+				var queryWithWhere = session.Query<Entity>()
+										.Where(WhereExpression(bob))
```

Comment:
Note for future - this indenting looks wrong depending on tab-size, because tab has been used also for alignment within a statement. Tabs are for indenting, space for alignment.